csm/core/services/usl.py

```diff
@@ -104,7 +105,48 @@ async def _get_system_friendly_name(self) -> str:
             raise CsmInternalError(desc=reason)
         return str(appliance_name)
 
-    def _get_device_uuid(self) -> UUID:
+    async def get_device_uuid(self) -> UUID:
+        return self._device_uuid
+
+    async def get_mgmt_url(self) -> str:
+        return ServiceUrls.get_mgmt_url()
+
+    async def get_public_ip(self) -> str:
+        return NetworkAddresses.get_node_public_data_ip_addr()
+
+    async def get_volumes(
```

Comment:
Minor doubt: can't USL interact with the S3 server directly?

Reply:
I believe it can in future versions.  Not directly in the literal sense, but the driver, instead of CSM itself, could be responsible for interacting with it.  For now, one of the things that hold it here is the storage capacity service; it would be too cumbersome to have it exposed in the driver as well.  Also, there is a chance that UDS itself will start managing devices directly, so we can hold this decision until we have a concrete idea if they are going to do it or not.